source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -362,12 +362,11 @@ Network::FilterStatus Filter::initializeUpstreamConnection() {
   }
 
   if (auto downstream_connection = downstreamConnection(); downstream_connection != nullptr) {
-    if (!read_callbacks_->connection()
-             .streamInfo()
+    if (!downstream_connection->streamInfo()
```

Comment:
nit, seems this can be `getStreamInfo` also.

Reply:
`getStreamInfo` will introduces some unnecessary calls `read_callbacks_->connection()` (although this overhead is almost negligible).